File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        
        for (int num : nums) {
            set.add(num);
        }

        int maxLen = 0;
        for (int num : set) {
            if (set.contains(num - 1)) {
                continue;
            }

            int len = 1;
            int cur = num;
            while (set.contains(cur + 1)) {
                len++;
                cur++;
            }

            maxLen = Math.max(maxLen, len);
        }

        return maxLen;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.kt
class Solution {
    fun longestConsecutive(nums: IntArray): Int {
        val set = nums.toSet()
        
        var maxLen = 0
        for (num in set) {
            if (num - 1 in set) {
                continue
            }

            var len = 1
            var cur = num
            while (cur + 1 in set) {
                len++
                cur++
            }

            maxLen = maxOf(maxLen, len)
        }

        return maxLen
    }
}